# Řešení druhé úlohy

## Ověřovaná hypotéza:
Pokud viník nehody byl pod silným vlivem alkoholu, došlo častěji k těžkým zdravotním následkům.


In [72]:
import numpy as np
import pandas as pd
import scipy.stats
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

Načtení dát a váběr užitečných sloupců 

In [73]:
df = pd.read_pickle("accidents.pkl.gz")

In [74]:
df = df[["p11", "p13a", "p13b"]]
df = df.dropna()

Binarizování hodnot 

In [79]:
df["silne_pod_vlivem"] = df["p11"] >= 7  # True pokud byl vynník pod vlivem alkoholu
df["ohrozeni_zivota"] = df["p13a"] + df["p13b"] > 0  # True pokud došlo k těžký zdravotním následkům

In [76]:
df

p11  p13a  p13b  silne_pod_vlivem  ohrozeni_zivota
0         2     0     0             False            False
1         2     0     0             False            False
2         2     0     0             False            False
3         9     0     0              True            False
4         0     0     0             False            False
...     ...   ...   ...               ...              ...
487156    0     0     0             False            False
487157    2     0     0             False            False
487158    0     0     0             False            False
487159    2     0     0             False            False
487160    0     0     0             False            False

[487161 rows x 5 columns]

Vytvoření kontingenční tabulky

In [77]:
crosstab = pd.crosstab(df["silne_pod_vlivem"], df["ohrozeni_zivota"])

In [78]:
crosstab

ohrozeni_zivota    False  True 
silne_pod_vlivem               
False             458884  10899
True               16492    886

In [81]:
chi2 = scipy.stats.chi2_contingency(crosstab)

In [84]:
chi2[1]  # vysledek

6.142715522048532e-121

# Závěr
Výsledek $\chi^2$ testu hypotéky je velmi nízký $6.1427e{-121}$, nulovou hypotézu tedy zamítáme a můžeme prohlásit, že mezi daty existuje koralace.